src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,27 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int MIN_AGE = 35;
+    private static final int MIN_PERIOD_IN_UKRAINE = 10;
+    private static final int LIVE_IN_UKR_FROM = 0;
+    private static final int LIVE_IN_UKR_TO = 1;
+    private static final String NATIONALITY = "Ukrainian";
+
+    @Override
+    public boolean test(Candidate candidate) {
+        return candidate != null
+                && candidate.getAge() >= MIN_AGE
+                && candidate.isAllowedToVote()
+                && candidate.getNationality().equals(NATIONALITY)
```

Comment:
```suggestion
                && REQUIRED_NATIONALITY.equals(candidate.getNationality())
```
this is how to avoid NPE in situations like this